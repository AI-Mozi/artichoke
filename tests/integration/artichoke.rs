use crate::run;

const BINARY: &str = "artichoke";
const FIXTURES_ROOT: &str = "./tests/integration/fixtures/";

#[test]
fn test_hello_world() {
    let app_name = "hello_world.rb";
    let path = format!("{}{}", FIXTURES_ROOT, app_name);
    insta::assert_toml_snapshot!(run(BINARY, &[&path]).unwrap());
}

#[test]
fn test_fizz_buzz() {
    let app_name = "fizz_buzz.rb";
    let path = format!("{}{}", FIXTURES_ROOT, app_name);
    insta::assert_toml_snapshot!(run(BINARY, &[&path]).unwrap());
}
