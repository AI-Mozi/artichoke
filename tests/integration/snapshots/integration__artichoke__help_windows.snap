---
source: tests/integration/artichoke.rs
assertion_line: 9
expression: "run(BINARY, &[\"--help\"]).unwrap()"

---
call_args = ['--help']
status = 0
stdout = '''
artichoke 0.1.0-pre.0
Artichoke is a Ruby made with Rust.

USAGE:
    artichoke.exe [FLAGS] [OPTIONS] [--] [programfile]...

FLAGS:
        --copyright    print the copyright
    -h, --help         Prints help information
    -V, --version      Prints version information

OPTIONS:
    -e <commands>...                one line of script. Several -e's allowed. Omit [programfile]
        --with-fixture <fixture>    file whose contents will be read into the `$fixture` global

ARGS:
    <programfile>...    
'''
stderr = ''
